### PR TITLE
Fix session storage for project name.

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
We were storing it just fine, but not initializing with it.

Closes https://github.com/microbit-foundation/python-editor-next/issues/547

### CODE REVIEW

**Comment by microbit-matt-hillsdon** [comment]
> Hmm, looks like e2e failure could be a real issue. Looking now.

Fixed in [db6a419](https://github.com/microbit-foundation/python-editor-next/pull/562/commits/db6a4198a1716e7d597e199c72ab0c405f73a4b3)

**Review by microbit-robert — APPROVED** [review]
LGTM.

File: src/fs/storage.ts [hunk 1]
```diff
@@ -93,11 +94,15 @@ export class SessionStorageFSStorage implements FSStorage {
   }
 
   async setProjectName(projectName: string) {
-    this.storage.setItem("projectName", projectName);
+    if (projectName === undefined) {
+      this.storage.removeItem(projectNameKey);
+    } else {
+      this.storage.setItem(projectNameKey, projectName);
+    }
```
Comment (microbit-matt-hillsdon) [inline src/fs/storage.ts:101]: Oops, I'd changed the signature to support undefined at some point, reverted it, and left this nonsense in.

```suggestion
    this.storage.setItem(projectNameKey, projectName);
```

File: src/fs/storage.ts [hunk 2]
```diff
@@ -225,12 +226,16 @@ export class SplitStrategyStorage implements FSStorage {
...
+const initializeFromStorage = async (from: FSStorage, to: FSStorage) => {
   const files = await from.ls();
-  return Promise.all(
-    files.map(async (f) => {
+  const projectName = await from.projectName();
+  if (projectName) {
+    await to.setProjectName(projectName);
+  }
+  return Promise.all([
+    ...files.map(async (f) => {
```
Comment (microbit-matt-hillsdon) [inline src/fs/storage.ts:236]: Oops, this is also odd, from when I updated project name unconditionally and so put it inside the Promise.all. The extra list is now pointless.

**Pull request merged**